x-pack/plugins/synthetics/public/legacy_uptime/components/overview/alerts/alert_query_bar/query_bar.tsx

```diff
@@ -53,6 +68,17 @@ export const AlertQueryBar = ({ query = '', onChange }: Props) => {
         placeholder={i18n.translate('xpack.synthetics.alerts.searchPlaceholder.kql', {
           defaultMessage: 'Filter using kql syntax',
         })}
+        appName={appName}
+        deps={{
+          unifiedSearch,
+          data,
+          storage,
+          notifications,
+          http,
+          docLinks,
+          uiSettings,
+          usageCollection,
```

Comment:
i think component should it self take care of these deps? i think it can be exported from the plugin contract, just like lens embeddable and and can inject all the deps it needs.